docs/graphs/classdiagram_lib.dot

```diff
@@ -0,0 +1,34 @@
+digraph hierarchy {
+size="5,5"
+node[   shape=record,
+        style=filled,
+        fillcolor=gray95]
+edge[dir=back, arrowtail=none]
+
+
+ps[label = "{ProgramState|
+                all_transactions\l
+                filtered_transactions\l|
+                + setup()\l
+                + add(dataframe)\l
+                + apply(filter)\l
+                + categorize(row, category)\l
+                + store()\l
+                }"]
+imp[label = "{TextImporter|| + read(file) : dataframe}"]
```

Comment:
1. Is the purpose of the `TextImporter` to import CSV files? If so, would the name `CSVImporter` be better?
1. I think it would be good to define an `Importer` interface, because many different kinds of importers are conceivable, like `CSVImporter`, `ExcelImporter`, `OnlineImporter`, ...
1. I think that such an importer should get all specific parameters like the `file` name in this example as a constructor argument and read the data by a generic method like `read() -> dataframe`. That way, only the creation of an importer is special for each kind (`CSVImporter` needs different constructor arguments than `OnlineImporter` for example), but reading from it would be generic.

Reply:
At first I thought of a classic approach like { AImporter <- CsvImporter <- DKBVisaImporter; CsvImporter <- SparkasseImporter} etc.
We can discuss this, but I find it's messy to have a different class for every importer. So at least for all text based formats, we could use one general importer, and all specifics are defined as *instances* of TextfileDescription. 
For every other importer an extra class. We can define everything that has a function `read(file) -> dataframe` to be a valid importer